driver/src/price_finding/naive_solver.rs

```diff
@@ -67,7 +67,9 @@ impl Matchable for Order {
     }
 
     fn have_price_overlap(&self, other: &Order) -> bool {
-        u128_to_u256(self.buy_amount) * u128_to_u256(other.buy_amount)
```

Comment:
This looks alright, but I find the long broken statement to be a little bit cosmetically unattractive. Maybe we can make an `all` and `any` macro at some point. I would love to learn how to make my own macros!